packages/opentelemetry-core/src/trace/Span.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { randomSpanId, randomTraceId } from '../platform';
+import { isValid } from './spancontext-utils';
+
+/**
+ * This class represents a span.
+ */
+export class Span implements types.Span {
+  private _traceId: string;
+  private _spanId: string;
+  private _parentId?: string;
+  private _traceState?: types.TraceState;
+  private _name: string;
+  private _kind: types.SpanKind;
+  private _attributes: types.Attributes = {};
+  private _links: types.Link[] = [];
+  private _events: types.Event[] = [];
+  private _status: types.Status = {
+    code: types.CanonicalCode.OK,
+  };
+  private _ended = false;
+  private _startTime: number;
+  private _duration = 0;
+
+  /** Constructs a new Span instance. */
+  constructor(
+    readonly _parentTracer: types.Tracer,
+    readonly _spanName: string,
+    readonly _options: types.SpanOptions
+  ) {
+    this._name = _spanName;
+    this._kind = _options.kind || types.SpanKind.INTERNAL;
+    this._spanId = randomSpanId();
+    const spanContext = this._getSpanContext(_options.parent);
+    if (spanContext && isValid(spanContext)) {
+      // New child span.
+      this._traceId = spanContext.traceId;
+      this._parentId = spanContext.spanId;
+      this._traceState = spanContext.traceState;
+    } else {
+      // This is a root span so no remote or local parent.
+      this._traceId = randomTraceId();
+    }
+    this._startTime = _options.startTime || performance.now();
+  }
+
+  tracer(): types.Tracer {
+    return this._parentTracer;
+  }
+
+  context(): types.SpanContext {
+    const spanContext: types.SpanContext = {
+      traceId: this._traceId,
+      spanId: this._spanId,
+      traceOptions: types.TraceOptions.SAMPLED,
```

Comment:
Why does this specify `SAMPLED` here?

Reply:
Also curious about this

Reply:
Was a little confuse here, Is it based on the sampling decision?

Reply:
Oh hm I guess this depends on the tracer's startSpan implementation. The basictracer implementation creates a noop span if the span isn't sampled, so the real span implementation here is always sampled: https://github.com/open-telemetry/opentelemetry-js/pull/99/files#diff-b4b335d4beecdee2577ffca9f9b398c7R84

Reply:
@draffensperger Do you agree on this?

Reply:
Ah, that makes sense. Could you just add a comment explaining that the non-sampled Spans use the NoOp span implementation, so that this will always be sampled?

Alternatively, can we have this be sent by the tracer itself so we don't need to set it here but rather do it in the constructor?